# Challenge 1

A and B are events of a certain probability space $(\omega, \sigma, P)$ such that $P(A) = 0.3$, $P(B) = 0.6$ and $P(A \cap B) = 0.1$.

Which are false?
* $P(A \cup B) = 0.6$
* $P(A \cap B^{C}) = 0.2$
* $P(A \cap (B \cup B^{C})) = 0.4$
* $P(A^{C} \cap B^{C}) = 0.3$
* $P((A \cap B)^{C}) = 0.9$





$P(A \cup B) = 0.6$ FALSE: 0.3 + 0.6 -0.1 = 0.8

$P(A \cap B^{C})$ = 0.2 TRUE: 0.3 - 0.1 = 0.2
   
$P(A \cap (B \cup B^{C}))$ = 0.4 FALSE: P(0.3 -1) = 0.7
    
$P(A^{C} \cap B^{C})$ = 0.3 TRUE : P(0.4 - 0.7) = 0.3
    
$P((A \cap B)^{C})$ = 0.9 TRUE: P(1-0.1) = 0.9

# Challenge 2
We have an urn with 10 white balls, 12 red balls and 8 black balls. Calculate the probability of:
* Taking a white ball.
* Taking a white ball after taking a black ball.
* Taking a red ball after taking a black and a red ball.
* Taking a red ball after taking a black and a red ball with reposition (that means after taking a ball I put it back to the urn).


P (Taking a white ball):

10/30 = 1/3

P(Taking a white ball after taking a black ball):

10/29

P(Taking a red ball after taking a black and a red ball):

11/28

P(Taking a red ball after taking a black and a red ball with reposition (that means after taking a ball I put it back to the urn)):

12/30 = 2/5



# Challenge 3

You are planning a picnic today, but the morning is cloudy. So you look for some data:
* 50% of all rainy days start off cloudy!
* But cloudy mornings are common (about 40% of days start cloudy)
* And this is usually a dry month (only 3 of 30 days (10%) tend to be rainy)

What is the chance of rain during the day?



P (cloudy|rain) = 0.5
P (cloudy) = 0.4
P (rain) = 0.10

P (rain|cloudy) = P (rain) * P(cloudy|rain)/P(cloudy) = 0.125

# Challenge 4

By conducting a telephone survey, we asked 1000 people if they thought it is necessary to have more street lighting at night.

We have met 480 men, 324 of which answered yes and 156 no, as well as 520 women, 351 of which answered yes and 169 no. We wonder if men and women have a different opinion or if gender is irrelevant to the question.

(To see more clearly what they told us, it is best to place the data in a table)


Consider the following events:
- Wants more light (answered yes)
- A man answered


We ask ourselves if these events are independent, that is, if the fact of wanting more light depends on whether one is male or female. Are these events independent or not?

In [5]:
import pandas as pd


In [16]:
pd.DataFrame({"Yes": [324/1000, 351/1000, (324/1000 + 351/1000)], "No": [156/1000, 169/1000, (156/1000 + 169/1000)], "Total": [(324/1000) + (156/1000), (351/1000) + (169/1000), (324/1000 + 351/1000) + (156/1000 + 169/1000)]}, index=["Men", "Women", "Total"])

Yes     No  Total
Men    0.324  0.156   0.48
Women  0.351  0.169   0.52
Total  0.675  0.325   1.00

The events are independent. 

P(Men) = 0.48

P(Yes) = 0.675

P(Yes|Man) = $P(Yes \cap Men )$ / P(Men)= 0.324/0.48 = 0.675 which is equal to P(Yes), this means the gender doesn't affect if they want more light or not, they are independent events. 